components/Table/Table.js

```diff
@@ -95,12 +113,184 @@ export class Table {
+              const dataElements = this.tableElementsMap[dataKey];
+              if (dataElements) {
+                this.tableElementsMap[dataKey].forEach(
+                  (item) => (item.style.display = 'none')
+                );
+              }
+              break;
+          }
+        });
+        return this.menuItem;
+      });
+
+    dropDownMenu.append(...this.menuDropDownItems);
+    menuBtn.append(dropDownMenu);
+
+    menuBtn.addEventListener('click', () => {
+      dropDownMenu.classList.toggle('show');
+    });
+  }
+
+  createFilterMenu() {
+    const filterMenuContainer = document.createElement('div');
+    filterMenuContainer.classList.add('filter-menu-container');
+
+    const filterLabel = document.createElement('div');
+    filterLabel.classList.add('filter-label');
+    filterLabel.textContent = 'Name';
+
+    const filterOperator = document.createElement('div');
+    filterOperator.classList.add('filter-operator');
+    filterOperator.textContent = 'equal';
+
+    const filterValue = document.createElement('input');
+    filterValue.classList.add('filter-value');
+    filterValue.placeholder = 'Filter value';
+
+    const btnClean = document.createElement('div');
+    btnClean.classList.add('btn-clean');
+    btnClean.textContent = 'x';
+
+    const btnExit = document.createElement('div');
+    btnExit.classList.add('btn-exit');
+    btnExit.textContent = 'x';
+
+    btnExit.addEventListener('click', () => {
+      filterMenuContainer.remove();
+    });
+
+    if (this.callbacksMap.sortCallback) {
+      btnClean.addEventListener('click', () => {
+        filterValue.value = '';
+        this.callbacksMap.sortCallback();
+      });
+    }
+
+    const filterFunction = (e) => {
+      let filter = e.target.value;
+      this.callbacksMap.filterCallback(filter);
+    };
+    filterValue.addEventListener('input', filterFunction);
+
+    filterMenuContainer.append(
+      btnClean,
+      filterLabel,
+      filterOperator,
+      filterValue,
+      btnExit
+    );
+    this.tableHeader.append(filterMenuContainer);
+  }
+
+  createPagination(totalAmount = 0) {
+    const paginationContainer = document.createElement('div');
+    paginationContainer.classList.add('pagination-container');
+
+    const dropdownPagintaionContainer = document.createElement('select');
+    dropdownPagintaionContainer.classList.add('dropdown-pagination-container');
+
+    this.pageNumberContainer = document.createElement('ul');
+    this.pageNumberContainer.classList.add('page-num-container');
+
+    const dropdownPaginationItems = PaginationConfigList.map((item) => {
+      const dropPagItem = document.createElement('option');
+      dropPagItem.classList.add('drop-pag-item');
+      dropPagItem.textContent = item.pageSize;
+
+      if (this.callbacksMap.paginationCallback) {
+        dropdownPagintaionContainer.addEventListener('change', (e) => {
+          if (this.amountElOnPage !== parseFloat(e.target.value)) {
```

Comment:
I guess here we should have `parseInt`, since we're working with numbers without dot